util/dvsim/NcLauncher.py

```diff
@@ -150,14 +157,51 @@ def poll(self):
         """
 
         assert self.process is not None
-        elapsed_time = datetime.datetime.now() - self.start_time
-        job_runtime_secs = elapsed_time.total_seconds()
         if self.process.poll() is None:
-            timeout_mins = self.deploy.get_timeout_mins()
-            if timeout_mins is not None and not self.deploy.gui:
-                if job_runtime_secs > (timeout_mins * 60):
+            run_timeout_mins = self.deploy.get_timeout_mins()
+            wait_timeout_mins = 180  # max wait time in job queue / license
+            file_size_thresh_bytes = 5000  # log file size threshold
+            if run_timeout_mins is not None and not self.deploy.gui:
+                job_waittime = 0
+                job_runtime = 0
+                # query the log file size
+                f_size = os.path.getsize(self.deploy.get_log_path())
+                if f_size < file_size_thresh_bytes:
+                    # if the job log size is less than the threshold, increment
+                    # the job wait time counter. hold the run time counter at 0
+                    self.start_run_time = 0
+                    elapsed_wait_time = (datetime.datetime.now() -
+                                         self.start_time)
+                    job_waittime = elapsed_wait_time.total_seconds() / 60
+                else:
+                    # if the job log size is more than the threshold, start the
+                    # the job run time counter if has not already started.
+                    # Hold the wait time counter at the last updated value
+                    if (self.start_run_time == 0):
+                        self.start_run_time = datetime.datetime.now()
```

Comment:
I'm not sure I properly understand this. Don't we want to pull this check to come at the start of this function, rather than sitting in the if/else block?

Reply:
I have introduced a new class variable nc_job_state which acts like a state machine and keeps track of current state in a more user friendly manner instead of having to keep track of the actual wait/run time 0 or non zero value and infer the job state. This allows to remove the duplicated code in areas you have mentioned above at the small cost of having a new class variable

Reply:
That seems like a good idea. Thanks!